### PR TITLE
V1.9.16 - Updates

### PR DESCRIPTION
**Pull request by ClutchplateDude**
- Added Meade extension command to move steppers by steps.

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -1,11 +1,19 @@
+; PlatformIO Project Configuration File
```
Comment (julianneswinoga) [inline platformio.ini]: Can you revert all of your changes to this file? I don't see any functional changes

File: src/Mount.hpp [hunk 2]
```diff
@@ -28,11 +28,13 @@ class TMC2209Stepper;
 #define TARGET_STRING      B01000
 #define CURRENT_STRING     B10000
 
-#define RA_STEPS  1
-#define DEC_STEPS 2
-#define AZIMUTH_STEPS 5
-#define ALTITUDE_STEPS 6
-#define FOCUS_STEPS 7
+enum StepperAxis {
```
Comment (julianneswinoga) [inline src/Mount.hpp:31]: Love it!! :heart: :+1: :+1: :+1:

File: src/Mount.cpp [hunk 3]
```diff
@@ -2790,6 +2817,16 @@ void Mount::calculateRAandDECSteppers(long& targetRASteps, long& targetDECSteps)
     float const RALimitR = (RA_LIMIT_LEFT * stepsPerSiderealHour);  
   #endif
 
+  if (pSolutions != NULL)
```
Comment (julianneswinoga) [inline src/Mount.cpp]: `nullptr` instead of `NULL`

File: src/Mount.cpp [hunk 4]
```diff
@@ -2790,6 +2817,16 @@ void Mount::calculateRAandDECSteppers(long& targetRASteps, long& targetDECSteps)
...
+  {
+    *(pSolutions + 0) = long(-moveRA);
```
Comment (julianneswinoga) [inline src/Mount.cpp]: mmmm don't do pointer arithmatic please. `pSolutions[0]`

File: src/Mount.cpp [hunk 5]
```diff
@@ -2740,7 +2767,7 @@ void Mount::calculateStepperPositions(float raCoord, float decCoord, long& raPos
 //
 // This code tells the steppers to what location to move to, given the select right ascension and declination
 /////////////////////////////////
-void Mount::calculateRAandDECSteppers(long& targetRASteps, long& targetDECSteps) const {
+void Mount::calculateRAandDECSteppers(long& targetRASteps, long& targetDECSteps, long* pSolutions) const {
```
Comment (julianneswinoga) [inline src/Mount.cpp]: I would do `long pSolutions[6]` to get compiler array length-checking

File: src/Mount.hpp [hunk 6]
```diff
@@ -411,7 +416,7 @@ class Mount {
   // Writes a 16-bit value to persistent (EEPROM) storage
   void writePersistentData(int which, long val);
 
-  void calculateRAandDECSteppers(long& targetRASteps, long& targetDECSteps) const;
+  void calculateRAandDECSteppers(long& targetRASteps, long& targetDECSteps, long* pSolutions = NULL) const;
```
Comment (julianneswinoga) [inline src/Mount.hpp]: `nullptr` here too

File: src/Mount.cpp [hunk 7]
```diff
@@ -2858,6 +2895,67 @@ void Mount::moveSteppersTo(float targetRASteps, float targetDECSteps) {   // Uni
   _stepperDEC->moveTo(targetDECSteps);
 }
 
+/////////////////////////////////
+//
+// moveSteppersTo
```
Comment (julianneswinoga) [inline src/Mount.cpp]: Fix doc name to match

**Review by julianneswinoga — CHANGES_REQUESTED** [review]

File: platformio.ini (hunk 1 above)
Comment (ClutchplateDude) [reply platformio.ini]: Can you explain how to do that after commit and push?

File: src/Mount.hpp (hunk 2 above)
Comment (ClutchplateDude) [reply src/Mount.hpp:31]: Knew you would, lol

File: src/Mount.cpp (hunk 4 above)
Comment (ClutchplateDude) [reply src/Mount.cpp]: If you insist. What's bad about pointer arithmetic?

File: src/Mount.cpp (hunk 5 above)
Comment (ClutchplateDude) [reply src/Mount.cpp]: Ohh, didn't even know that's legal syntax. Fixed.

File: src/Mount.cpp (hunk 7 above)
Comment (ClutchplateDude) [reply src/Mount.cpp]: Good catch. Fixed.

File: src/Mount.cpp (hunk 4 above)
Comment (julianneswinoga) [reply src/Mount.cpp]: It's error prone and can lead to some nasty bugs. 99.9% of the time you can accomplish what you need with code that can be length-checked at compile time

File: platformio.ini (hunk 1 above)
Comment (ClutchplateDude) [reply platformio.ini]: OK, reverted.

File: src/Mount.cpp (hunk 3 above)
Comment (ClutchplateDude) [reply src/Mount.cpp]: Fixed

File: src/Mount.cpp (hunk 4 above)
Comment (ClutchplateDude) [reply src/Mount.cpp]: Fixed

File: src/Mount.hpp (hunk 6 above)
Comment (ClutchplateDude) [reply src/Mount.hpp]: Fixed.

**Review by julianneswinoga — APPROVED** [review]

**Pull request merged**